neonvm/daemon/Dockerfile

```diff
@@ -0,0 +1,11 @@
+ARG GO_BASE_IMG=autoscaling-go-base:dev
+FROM $GO_BASE_IMG AS builder
```

Comment:
Worth considering: We probably _shouldn't_ use go-chef here (and as a dependency in the makefile), because otherwise the `build-test-vm` workflow will have to build all dependencies if there's a cache miss.

And, neonvm-daemon should be pretty quick to compile anyways :)